### PR TITLE
Platform checkout: Remove WooCommerce Blocks plugin dependency

### PR DESCRIPTION
**Pull request by ricardo**
Context: p1650319227659789-slack-C022WMN88KG

#### Changes proposed in this Pull Request

This injects the necessary payments API classes only available behind constant `WC_BLOCKS_IS_FEATURE_PLUGIN`, so they're available in core for the platform checkout.

#### Testing instructions

* Make sure the WooCommerce Blocks plugin is disabled.
* As a shopper, add a product to the cart.
* Proceed to checkout and enter a platform checkout user email.
* Proceed with the platform checkout, and ensure you're redirected back to the "Order received" page.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_

### CODE REVIEW

File: includes/class-wc-payments.php [hunk 1]
```diff
@@ -935,6 +935,22 @@ function () {
...
+				$blocks_package_container = Automattic\WooCommerce\Blocks\Package::container();
+				$blocks_package_container->register(
+					Automattic\WooCommerce\Blocks\Payments\Api::class,
+					function ( $container ) {
+						$payment_method_registry = $container->get( Automattic\WooCommerce\Blocks\Payments\PaymentMethodRegistry::class );
+						$asset_data_registry     = $container->get( Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry::class );
+						return new Automattic\WooCommerce\Blocks\Payments\Api( $payment_method_registry, $asset_data_registry );
+					}
+				);
+				$blocks_package_container->get( Automattic\WooCommerce\Blocks\Payments\Api::class );
```
Comment (Mayisha) [inline includes/class-wc-payments.php:952]: Is this needed here? :thinking:

**Review by Mayisha — COMMENTED** [review]
Tested for these combinations-
- WC 5.2, WC 5.4
- With and without Blocks
- Platform checkout enabled and disabled

Worked for all the combinations.

File: includes/class-wc-payments.php (hunk 1 above)
Comment (ricardo) [reply includes/class-wc-payments.php:952]: You mean this line?

```php
$blocks_package_container->get( Automattic\WooCommerce\Blocks\Payments\Api::class );
```

It replaces [this line](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/src/Domain/Bootstrap.php#L103) from the Woo Blocks plugin, and needs to happen after [this](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/src/Domain/Bootstrap.php#L265-L272).

Comment (Mayisha) [reply includes/class-wc-payments.php:952]: Got it!

File: includes/class-wc-payments.php [hunk 2]
```diff
@@ -935,6 +935,22 @@ function () {
 					}
 				}
 			);
+
+			// This injects the payments API into core, so the WooCommerce Blocks plugin is not necessary.
+			// The payments API is currently only available in feature builds (with flag `WC_BLOCKS_IS_FEATURE_PLUGIN`).
+			// We should remove this once it's available in core by default.
+			if ( ! defined( 'WC_BLOCKS_IS_FEATURE_PLUGIN' ) && class_exists( 'Automattic\WooCommerce\Blocks\Payments\Api' ) ) {
```
Comment (Mayisha) [inline includes/class-wc-payments.php:942]: As discussed over the [thread](p1650642013069719/1650319227.659789-slack-C022WMN88KG) let's add a condition here to check if `woopay` is eligible and enabled.

Comment (ricardo) [reply includes/class-wc-payments.php:942]: We're doing it right above on line 924 🙂.

Comment (Mayisha) [reply includes/class-wc-payments.php:942]: My bad. It seemed to me outside of that `if` check :facepalm: 
Sorry for the confusions :facepalm:

Comment (ricardo) [reply includes/class-wc-payments.php:942]: It's easy to miss that since GH doesn't display the full scope of where the diff is 🙂.

**Comment by asumaran** [comment]
Tests well.

Tested with:

WC 6.4.1
WC Blocks 7.4.2
WC Payments 4.0.1

With or without WC Blocks enabled.

**Review by asumaran — APPROVED** [review]
Tests well.

Tested with:

WC 6.4.1
WC Blocks 7.4.2
WC Payments 4.0.1

With or without WC Blocks enabled.

**Review by Mayisha — APPROVED** [review]
Looks good :tada: 

Tested these scenarios and they worked fine.

WC 6.4.1
- without Blocks, platform checkout enabled
   - woopay, card payment, cod
- without Blocks, platform checkout disabled
   - card payment, cod
- with Blocks 7.4.2, platform checkout enabled
    - woopay, card payment, cod, blocks checkout
- with Blocks 7.4.2, platform checkout disabled
    - card payment, cod, blocks checkout

WC 5.2.0
- without Blocks, platform checkout enabled
   - woopay, card payment, cod
- without Blocks, platform checkout disabled
   - card payment, cod
- with Blocks 5.5.0, platform checkout enabled
    - woopay, card payment, cod, blocks checkout
- with Blocks 5.5.0, platform checkout disabled
    - card payment, cod, blocks checkout

**Pull request merged**